infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -339,11 +339,27 @@ private synchronized void runDetectionTasks() {
             // Corrects out of phase epoch issues if present in the report. This method
             // performs re-sealing of all nodes if required and catchup of a layout server to
             // the current state.
-            Layout updatedLayout = correctWrongEpochs(pollReport, layout);
+            final Layout updatedLayout = correctWrongEpochs(pollReport, layout);
 
             Result<DetectorTask, RuntimeException> failure = Result.of(() -> {
-                if (pollReport.isCurrentLayoutSlotUnFilled()) {
-                    detectFailure(updatedLayout, Collections.emptySet(), pollReport).join();
+
+                // This is just an optimization in case we receive a WrongClusterException
+                // while one of the other management clients is trying to move to a new layout.
+                // This check is merely trying to minimize the the scenario in which we end ip
+                // filling the slot with an outdated layout. Due to potential failures, we cannot
+                // wait for more than the majority of layout servers.
+                if (!pollReport.isMajorityLayoutServersSealed(updatedLayout)) {
+                    log.info(("Majority of the layout servers have not been sealed. Skipping.");
+                    return DetectorTask.COMPLETED;
+                }
+
+                Optional<Long> unfilledSlot = pollReport.getLayoutSlotUnFilled(updatedLayout);
+                // If the latest slot has not been filled, fill it with the previous known layout.
+                if (unfilledSlot.isPresent()) {
+                    log.info("Trying to fill an unfilled slot {}.", unfilledSlot.get());
```

Comment:
looks like we need more info in the log, can you please add `pollReport` into the log

Reply:
Done.